example/spec/ingest/ingestFromPdrSpec.js

```diff
@@ -0,0 +1,441 @@
+  waitForCompletedExecution,
+  LambdaStep,
+  addProviders,
+  cleanupProviders,
+  addCollections,
+  cleanupCollections,
+  api: apiTestUtils
+} = require('@cumulus/integration-tests');
+
+const {
+  loadConfig,
+  updateAndUploadTestDataToBucket,
+  deleteFolder,
+  createTimestampedTestId,
+  createTestDataPath,
+  createTestSuffix,
+  getExecutionUrl
+} = require('../helpers/testUtils');
+
+const {
+  loadFileWithUpdatedGranuleIdPathAndCollection
+} = require('../helpers/granuleUtils');
+
+const config = loadConfig();
+const lambdaStep = new LambdaStep();
+const taskName = 'DiscoverAndQueuePdrs';
+const pdrFilename = 'MOD09GQ_1granule_v3.PDR';
+
+const s3data = [
+  '@cumulus/test-data/pdrs/MOD09GQ_1granule_v3.PDR',
+  '@cumulus/test-data/granules/MOD09GQ.A2016358.h13v04.006.2016360104606.hdf.met',
+  '@cumulus/test-data/granules/MOD09GQ.A2016358.h13v04.006.2016360104606.hdf'
+];
+
+describe('Ingesting from PDR', () => {
+  const testId = createTimestampedTestId(config.stackName, 'IngestFromPdr');
+  const testSuffix = createTestSuffix(testId);
+  const testDataFolder = createTestDataPath(testId);
+  const providersDir = './data/providers/s3/';
+  const collectionsDir = './data/collections/s3_MOD09GQ_006';
+  const collection = { name: `MOD09GQ${testSuffix}`, version: '006' };
+  const provider = { id: `s3_provider${testSuffix}` };
+  process.env.ExecutionsTable = `${config.stackName}-ExecutionsTable`;
+  process.env.CollectionsTable = `${config.stackName}-CollectionsTable`;
+  process.env.PdrsTable = `${config.stackName}-PdrsTable`;
+  const executionModel = new Execution();
+  const collectionModel = new Collection();
+  const pdrModel = new Pdr();
+  let parsePdrExecutionArn;
+
+  beforeAll(async () => {
+    // delete pdr from old tests
+    await apiTestUtils.deletePdr({
+      prefix: config.stackName,
+      pdr: pdrFilename
+    });
+
+    // populate collections, providers and test data
+    await Promise.all([
+      updateAndUploadTestDataToBucket(config.bucket, s3data, testDataFolder, [{ old: 'cumulus-test-data/pdrs', new: testDataFolder }, { old: 'DATA_TYPE = MOD09GQ;', new: `DATA_TYPE = MOD09GQ${testSuffix};` }]),
+      addCollections(config.stackName, config.bucket, collectionsDir, testSuffix),
+      addProviders(config.stackName, config.bucket, providersDir, config.bucket, testSuffix)
+    ]);
+    // update provider path
+    await collectionModel.update(collection, { provider_path: testDataFolder });
+  });
+
+  afterAll(async () => {
+    // clean up stack state added by test
+    await Promise.all([
+      deleteFolder(config.bucket, testDataFolder),
+      cleanupCollections(config.stackName, config.bucket, collectionsDir, testSuffix),
+      cleanupProviders(config.stackName, config.bucket, providersDir, testSuffix),
+      apiTestUtils.deletePdr({
+        prefix: config.stackName,
+        pdr: pdrFilename
+      })
+    ]);
+  });
+
+  describe('The Discover and Queue PDRs workflow', () => {
```

Comment:
Since there is only one `describe` statement (this one) inside of the describe statement starte online 61, could they just be consolidated into one describe statement?  That would also mean that you could combine the `beforeAll` blocks.

Reply:
I don't know what that buys us, I think it makes the one beforeAll extra confusing because you'd be defining lots of variables that get used way down the road. Having it split up makes these long tests more manageable and readable and actually scopes the variables.